### PR TITLE
refactor: fixed go lint issues

### PR DESCRIPTION
**Pull request by baizon**
- Update `bitboxswiss/bitbox-wallet-app-ci` docker image to version `23`
- Update `.golangci.yml` to be compatible with version `1.59.0`
- Fixed all the go lint issues
- Added `golangci-lint config verify` to `ci.sh` to check the go lint configuration (`.golangci.yml`)

### CODE REVIEW

File: backend/rates/history.go [hunk 1]
```diff
@@ -178,7 +178,8 @@ func (updater *RateUpdater) updateHistory(ctx context.Context, coin, fiat string
 
 	updater.historyMu.Lock()
 	defer updater.historyMu.Unlock()
-	allRates := append(updater.history[bucketName], fetchedRates...)
+	updater.history[bucketName] = append(updater.history[bucketName], fetchedRates...)
+	allRates := updater.history[bucketName]
 	sort.Slice(allRates, func(i, j int) bool {
 		return allRates[i].timestamp.Before(allRates[j].timestamp)
 	})
```
Comment (benma) [inline backend/rates/history.go:185]: the line below this one is not needed I think. For clarity maybe it's better to just do this?

```
	updater.history[bucketName] = append(updater.history[bucketName], fetchedRates...)
	sort.Slice(updater.history[bucketName], func(i, j int) bool {
		return updater.history[bucketName][i].timestamp.Before(updater.history[bucketName][j].timestamp)
	})
	return len(fetchedRates), nil
```

wdyt?

File: backend/rates/history_test.go [hunk 2]
```diff
@@ -42,9 +42,7 @@ func TestPriceAt(t *testing.T) {
 		{7.25, time.Date(2020, 9, 3, 18, 0, 0, 0, time.UTC)},
 	}
 	for _, test := range tt {
-		assert.Equal(t,
-			test.wantValue,
-			updater.HistoricalPriceAt("btc", "USD", test.at), "at = %s", test.at)
+		assert.InDelta(t, test.wantValue, updater.HistoricalPriceAt("btc", "USD", test.at), 0, "at = %s", test.at)
```
Comment (benma) [inline backend/rates/history_test.go:45]: This is an instance of making code less readable to please the linter. Can the `InDelta` check only be disabled?

**Review by benma — CHANGES_REQUESTED** [review]
Very nice, thanks a lot! I left two small nits.

File: backend/rates/history.go (hunk 1 above)
Comment (baizon) [reply backend/rates/history.go:185]: Yes, this is a better solution imho. Also the reassignment `updater.history[bucketName] = allRates` is not useful and just more memory association. I've applied your change.

**Review by baizon — COMMENTED** [review]
Build in review comments.

File: backend/rates/history_test.go [hunk 3]
```diff
@@ -42,7 +42,10 @@ func TestPriceAt(t *testing.T) {
 		{7.25, time.Date(2020, 9, 3, 18, 0, 0, 0, time.UTC)},
 	}
 	for _, test := range tt {
-		assert.InDelta(t, test.wantValue, updater.HistoricalPriceAt("btc", "USD", test.at), 0, "at = %s", test.at)
+		//nolint:testifylint
```
Comment (benma) [inline backend/rates/history_test.go:45]: There are 4 other instances of InDelta that you could revert.

**Review by benma — CHANGES_REQUESTED** [review]

File: backend/rates/history_test.go (hunk 3 above)
Comment (baizon) [reply backend/rates/history_test.go:45]: A, ok, i thought you mean it just for that one case. I'll revert it and exclude the rule globally.

**Review by benma — COMMENTED** [review]
Thanks a lot. Please squash all commits, then we can merge.

**Comment by baizon** [comment]
Done.

**Review by benma — APPROVED** [review]
LGTM thanks!

**Pull request merged**